src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.cpp

```diff
@@ -939,7 +939,7 @@ void ControlBoardWrapper::run()
 //
 bool ControlBoardWrapper::setPid(int j, const Pid &p)
 {
-    int off=device.lut[j].offset;
+    int off; try{off = device.lut.at(j).offset;} catch(...){yError() << "joint number " << j <<  " out of bound [0-"<< controlledJoints << "] for part " << partName; return false; }
```

Comment:
The idea is that we are checking for something that should never occour in normal usage
In thise case, the try/catch mechanism is more efficent than a double `if()`  (double because we need to check 2 bounds).

Exceptions are slower than normal `if()` code only when actually thrown, which is rare and anyway better than a segfault :-)
http://stackoverflow.com/questions/13835817/are-exceptions-in-c-really-slow